### ISSUE TITLE
Detect invalid varargs signature

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#1976]
A varargs signature can only have one remaining argument. E.g. the following is invalid:

``` clojure
(defn when-pred [pred & x args]
  (try (when (apply pred x args)
         x)
       (catch Exception _ nil)))
```

This should be detected by the `:syntax` linter.
